app/controllers/topics_controller.rb

```diff
@@ -90,6 +110,10 @@ def destroy
 
   private
 
+  def get_room
```

Comment:
I'd name this one `set_room`, too, for consistency with `set_room` on your rooms controller.